src/docs/usr-manual/ch-sstruct.rst

```diff
@@ -79,78 +73,72 @@ Consider the solution of the diffusion equation
-separate logically-rectangular parts as shown in Figure
-:ref:`fig-sstruct-example-parts`, and each part is given a unique label between
-0 and 4.  Each part consists of a single box with lower index :math:`(1,1)` and
-upper index :math:`(4,4)` (see Section :ref:`sec-Struct-Grid`), and the grid
-data is distributed on five processes such that data associated with part
-:math:`p` lives on process :math:`p`.  Note that in general, parts may be
-composed out of arbitrary unions of boxes, and indices may consist of
-non-positive integers (see Figure :ref:`fig-struct-boxes`).  Also note that the
-``SStruct`` interface expects a domain-based data distribution by boxes, but the
-actual distribution is determined by the user and simply described (in parallel)
-through the interface.
-
-.. |figSStructGrid1| image:: figSStructGrid1.*
+   :name: fig-sstruct-example-parts
+
+   : One possible labeling of the grid in :numref:`fig-sstruct-example`.
+
+The grid in :numref:`fig-sstruct-example` is defined in terms of five separate
+logically-rectangular parts as shown in :numref:`fig-sstruct-example-parts`, and each part
+is given a unique label between 0 and 4.  Each part consists of a single box with lower
+index :math:`(1,1)` and upper index :math:`(4,4)` (see Section :ref:`sec-Struct-Grid`),
+and the grid data is distributed on five processes such that data associated with part
+:math:`p` lives on process :math:`p`.  Note that in general, parts may be composed out of
+arbitrary unions of boxes, and indices may consist of non-positive integers (see
+:numref:`fig-struct-boxes`).  Also note that the ``SStruct`` interface expects a
+domain-based data distribution by boxes, but the actual distribution is determined by the
+user and simply described (in parallel) through the interface.
+
+.. |figSStructGrid1| image:: figSStructGrid1.svg
    :width: 100%
-.. |figSStructGrid2| image:: figSStructGrid2.*
+.. |figSStructGrid2| image:: figSStructGrid2.svg
    :width: 100%
-.. |figSStructGrid3| image:: figSStructGrid3.*
+.. |figSStructGrid3| image:: figSStructGrid3.svg
    :width: 100%
-.. |figSStructGrid4| image:: figSStructGrid4.*
+.. |figSStructGrid4| image:: figSStructGrid4.svg
    :width: 100%
-.. |figSStructGrid5| image:: figSStructGrid5.*
+.. |figSStructGrid5| image:: figSStructGrid5.svg
    :width: 100%
-.. |figSStructGrid6| image:: figSStructGrid6.*
+.. |figSStructGrid6| image:: figSStructGrid6.svg
    :width: 100%
 
 .. _fig-sstruct-grid:
-    
-+----------------------+----------------------+----------------------+
-| 1: |figSStructGrid1| | 2: |figSStructGrid2| | 3: |figSStructGrid3| |
-+----------------------+----------------------+----------------------+ 
-| 4: |figSStructGrid4| | 5: |figSStructGrid5| | 6: |figSStructGrid6| |
-+----------------------+----------------------+----------------------+
+
+.. table:: : Grid Setup Process. The "icons" illustrate the result of the numbered lines of code.
```

Comment:
I'm not sure I like the table approach here.  This together with the following code block is really supposed to be another figure.  In the structured grid section, we don't label these at all, just let them flow with the text.  Maybe we should just keep this as before and fix the reference below (I'll mark it now)?

Reply:
With my new numbering scheme (`numref`) I need to add a caption to this to be able to cite it in the text. So I can't have this as before (master). Hope this is fine